jax/_src/ffi.py

```diff
@@ -15,9 +15,48 @@
 from __future__ import annotations
 
 import os
+import ctypes
 
 from jax._src.lib import jaxlib
 
+# Use Python stable C API function to construct a PyCapsule.
+# From ctypes.pythonapi documentation:
+#
+#  ctypes.pythonapi
+#    An instance of PyDLL that exposes Python C API functions as attributes.
+#    Note that all these functions are assumed to return C int, which is of
+#    course not always the truth, so you have to assign the correct restype
+#    attribute to use these functions.
+#
+# Taken from: https://docs.python.org/3/library/ctypes.html.
+#
+# More context here: https://stackoverflow.com/questions/24377845/ctype-why-specify-argtypes.
+#
+# Following this advice we annotate argument and return types of PyCapsule_New
+# before we call it. Example here:
+# https://stackoverflow.com/questions/65056619/converting-ctypes-c-void-p-to-pycapsule
+def build_capsule(funcptr):
```

Comment:
Maybe just `pycapsule` (`py_capsule`) or `capsule`?

Reply:
Renamed to `pycapsule`.